source/parser/parser.cpp

```diff
@@ -4892,51 +4894,115 @@ ObjectPtr Parser::Parse_Ovus()
     Object->BottomRadius = Parse_Float(); /* Bottom radius */
     Parse_Comma();
     Object->TopRadius = Parse_Float(); /* Top radius */
+    // default values, for backward compatibility
+    Object->VerticalSpherePosition = Object->BottomRadius;
+    Object->ConnectingRadius = 2.0 * std::max(Object->BottomRadius, Object->TopRadius);
+    EXPECT
+        CASE(RADIUS_TOKEN)
+            Object->ConnectingRadius = Parse_Float();
+        END_CASE
+        CASE(DISTANCE_TOKEN)
+            Object->VerticalSpherePosition = Parse_Float();
+        END_CASE
+        CASE(PRECISION_TOKEN)
+            Object->RootTolerance = Parse_Float();
+        END_CASE
+        OTHERWISE
+            UNGET
+            EXIT
+        END_CASE
+    END_EXPECT
 
     /*
      ** Pre-compute the important values
      */
-    if ((Object->TopRadius < 0)||(Object->BottomRadius < 0))
+    if ((Object->TopRadius < 0)||(Object->BottomRadius <= 0))
     {
-        Error("Both Ovus radii must be positive");
+        Error("Both Ovus radii must be positive, and bottom radius must be strictly positive");
     }
-    if (Object->TopRadius < 2.0 * Object->BottomRadius)
+    if (Object->VerticalSpherePosition < Object->BottomRadius)
     {
-        if (Object->BottomRadius > Object->TopRadius)
-        {
-            Object->ConnectingRadius = 2.0 * Object->BottomRadius;
-            Object->VerticalPosition = 2.0 * Object->TopRadius - Object->BottomRadius - (Object->TopRadius * Object->TopRadius / (2.0 * Object->BottomRadius) );
-            Object->HorizontalPosition = sqrt((Object->BottomRadius)*(Object->BottomRadius) -(Object->VerticalPosition)*(Object->VerticalPosition));
-            Object->BottomVertical = -Object->VerticalPosition;
-            distance = Object->ConnectingRadius - Object->TopRadius;
-            Object->TopVertical = ((Object->BottomRadius - Object->VerticalPosition) * Object->TopRadius / distance ) + Object->BottomRadius;
-        } else {
-            Object->ConnectingRadius = 2.0 * Object->TopRadius;
-            Object->VerticalPosition = - 2.0 * Object->TopRadius + Object->BottomRadius + (1.5 * Object->TopRadius * Object->TopRadius / Object->BottomRadius);
-            Object->HorizontalPosition = sqrt((Object->TopRadius)*(Object->TopRadius) - ((Object->VerticalPosition - Object->BottomRadius) * (Object->VerticalPosition - Object->BottomRadius)));
-            Object->TopVertical = 2.0 * Object->BottomRadius - Object->VerticalPosition;
-            distance = Object->ConnectingRadius - Object->BottomRadius;
-            Object->BottomVertical = -Object->VerticalPosition * Object->BottomRadius / distance;
-        }
+        Error("distance of Ovus must be greater than bottom radius");
```

Comment:
Why? Shouldn't the ovus object also work with a lower VSP, provided VSP+TR > BR?